src/prefect/core/flow.py

```diff
@@ -1472,15 +1480,22 @@ def serialized_hash(self, build: bool = False) -> str:
         determining if the flow has changed. If this hash is equal to a previous hash,
         no new information would be passed to the server on a call to `flow.register()`
 
+        Note that this will not always detect code changes since the task code is not
+        included in the serialized flow sent to the server. That said, as long as the
+        flow is "built" during registration, the code changes will be in effect even
+        if a new version is not registered with the server.
+
         Args:
             - build (bool, optional):  if `True`, the flow's environment is built
                 prior to serialization. Passed through to `Flow.serialize()`.
 
         Returns:
             - str: the hash of the serialized flow
         """
+        serialized_flow = self.serialize(build)
+
         return hashlib.sha256(
-            json.dumps(self.serialize(build), sort_keys=True).encode()
+            json.dumps(serialized_flow, sort_keys=True).encode()
```

Comment:
This pulled out because I was transforming the serialized_flow but switched to transforming the flow copy before serialization. Still useful pulled out for debugging.

Reply:
Question - instead of messing around with the serialization logic, couldn't you do:
```python
"".join(sorted(json.dumps(serialized_flow)).encode()
```
which would detect any changes?

Reply:
This PR would work it seems, but will be really hard to maintain as the changes are so diverse

Reply:
I did try sorting after serializing but it's a bit of a pain. The issue here are there are lists of dictionaries and the dictionaries cannot be sorted in the list without identifying which key you'd like to sort by. I think it's more clear to sort at the schema level and actually I think if I move the sorting next to the `FlowSchema` definition and add a note that nested schemas with `many=True` will need to be sorted for determinism it'll be pretty easy to maintain.

Reply:
Chris just schooooooled me. I'll be updating this with that recommendation and a comment.